x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/service/DeleteServiceAccountTokenRequest.java

```diff
@@ -0,0 +1,117 @@
+        this.serviceName = in.readString();
+        this.tokenName = in.readString();
+        this.refreshPolicy = WriteRequest.RefreshPolicy.readFrom(in);
+    }
+
+    public String getNamespace() {
+        return namespace;
+    }
+
+    public String getServiceName() {
+        return serviceName;
+    }
+
+    public String getTokenName() {
+        return tokenName;
+    }
+
+    public WriteRequest.RefreshPolicy getRefreshPolicy() {
+        return refreshPolicy;
+    }
+
+    public void setRefreshPolicy(WriteRequest.RefreshPolicy refreshPolicy) {
+        this.refreshPolicy = Objects.requireNonNull(refreshPolicy, "refresh policy may not be null");
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        DeleteServiceAccountTokenRequest that = (DeleteServiceAccountTokenRequest) o;
+        return Objects.equals(namespace, that.namespace) && Objects.equals(serviceName, that.serviceName)
+            && Objects.equals(tokenName, that.tokenName) && refreshPolicy == that.refreshPolicy;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(namespace, serviceName, tokenName, refreshPolicy);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(namespace);
+        out.writeString(serviceName);
+        out.writeString(tokenName);
+        refreshPolicy.writeTo(out);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (Strings.isNullOrEmpty(namespace)) {
+            validationException = addValidationError("service account namespace is required", validationException);
+        }
+
+        if (Strings.isNullOrEmpty(serviceName)) {
+            validationException = addValidationError("service account service-name is required", validationException);
+        }
+
+        if (Strings.isNullOrEmpty(tokenName)) {
+            validationException = addValidationError("service account token name is required", validationException);
+        } else {
+            if (tokenName.length() > 256) {
+                validationException = addValidationError(
+                    "service account token name may not be more than 256 characters long", validationException);
+            }
+            if (tokenName.equals(tokenName.trim()) == false) {
+                validationException = addValidationError(
+                    "service account token name may not begin or end with whitespace", validationException);
+            }
+            if (tokenName.startsWith("_")) {
+                validationException = addValidationError(
+                    "service account token name may not begin with an underscore", validationException);
+            }
+        }
```

Comment:
However, merging this validation does make sense, and would require having some sort of common interface (or a common `TokenIdentifier` field).

Reply:
Yes it should be merged. In fact, it should be replaced by ServiceAccountToken#isValidTokenName
https://github.com/elastic/elasticsearch/blob/3d784f8b14fcb6421c7914e8f9d4ffc8ba72956d/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ServiceAccountToken.java#L172-L174

But I didn't do it because `ServiceAccountToken` is in the `security` package and not accessible here. I guess this is the forgotten reason that I initially moved it from `security` to `core`. But because the 3rd PR didn't need it then I forgot about why and moved it back to `security`. Now it seems we should move it again to `core`. It can be done easily, but will touch a lot files with trivial changes. Are you OK with it?

Reply:
That's fine. The other alternative is to move the method to `org.elasticsearch.xpack.core.security.support.Validation`

Reply:
I moved the validation method into `org.elasticsearch.xpack.core.security.support.Validation` and also associated tests as well. It also touches a few other files. But the changes are trivial.